### PR TITLE
Add h2spec test suite module to check if netty http2 implementation conforms with the specification

### PR DESCRIPTION
**Pull request by madgnome**
Motivation:

H2Spec is a conformance testing tool for HTTP/2 implementation.
To help us fix failing tests and avoid future regression we
should run h2spec as part of the build

Modifications:

- Add testsuite-http2 module to the project

Result:

- Run h2spec as part of the build
- 22 tests are currently ignored, we should remove the ignore as we fix them
- Automatic tests for #5761

### CODE REVIEW

File: testsuite-http2/pom.xml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  <profiles>
+    <profile>
+      <id>skipTests</id>
+      <activation>
+        <property>
+          <name>skipTests</name>
+        </property>
+      </activation>
+      <properties>
+        <skipAutobahnTestsuite>true</skipAutobahnTestsuite>
```
Comment (normanmaurer) [inline testsuite-http2/pom.xml]: I think this should be `skipHttp2Testsuite`

File: testsuite-http2/src/main/java/io/netty/testsuite/http2/HelloWorldHttp1Handler.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2014 The Netty Project
```
Comment (normanmaurer) [inline testsuite-http2/src/main/java/io/netty/testsuite/http2/HelloWorldHttp1Handler.java]: 2017

File: testsuite-http2/src/main/java/io/netty/testsuite/http2/HelloWorldHttp2Handler.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright 2014 The Netty Project
```
Comment (normanmaurer) [inline testsuite-http2/src/main/java/io/netty/testsuite/http2/HelloWorldHttp2Handler.java]: 2017

File: testsuite-http2/src/main/java/io/netty/testsuite/http2/HelloWorldHttp2HandlerBuilder.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2015 The Netty Project
```
Comment (normanmaurer) [inline testsuite-http2/src/main/java/io/netty/testsuite/http2/HelloWorldHttp2HandlerBuilder.java]: 2017

File: testsuite-http2/src/main/java/io/netty/testsuite/http2/Http2Server.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2014 The Netty Project
```
Comment (normanmaurer) [inline testsuite-http2/src/main/java/io/netty/testsuite/http2/Http2Server.java]: 2017

File: testsuite-http2/src/main/java/io/netty/testsuite/http2/Http2ServerInitializer.java [hunk 6]
```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2014 The Netty Project
```
Comment (normanmaurer) [inline testsuite-http2/src/main/java/io/netty/testsuite/http2/Http2ServerInitializer.java]: 2017

File: testsuite-http2/src/main/java/io/netty/testsuite/http2/package-info.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * This package is intended to test the http2 implementation against the specification
+ * using h2spec
+ */
+package io.netty.testsuite.http2;
+
```
Comment (normanmaurer) [inline testsuite-http2/src/main/java/io/netty/testsuite/http2/package-info.java]: remove empty line

File: testsuite-http2/src/main/java/io/netty/testsuite/http2/package-info.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2012 The Netty Project
```
Comment (normanmaurer) [inline testsuite-http2/src/main/java/io/netty/testsuite/http2/package-info.java]: 2017

**Comment by normanmaurer** [comment]
@carl-mastrangelo @ejona86 @Scottmitch WDYT ? I think its a nice one :)

**Comment by normanmaurer** [comment]
Squashed and cherry-picked into 4.1 (454868654427433cd496c9f172615abb1b2b4bd5). This is super awesome and helpful!

**Comment by Scottmitch** [comment]
concept sound good to me, but there seems to be a lot of duplication between this code and the examples code. Is there anyway to remove the duplication and consolidate the code?

**Pull request closed without merging**